manager_test.go

```diff
@@ -35,165 +35,176 @@ func (m *customMid) Trace() []string {
+	rc := Config.Client
 
+	processed := make(chan *Args)
+	testJob := (func(message *Msg) {
+		processed <- message.Args()
 	})
 
-	c.Specify("manage", func() {
-		rc := Config.Client
-
-		message, _ := NewMsg("{\"foo\":\"bar\",\"args\":[\"foo\",\"bar\"]}")
-		message2, _ := NewMsg("{\"foo\":\"bar2\",\"args\":[\"foo\",\"bar2\"]}")
+	manager := newManager("manager1", testJob, 10)
 
-		c.Specify("coordinates processing of queue messages", func() {
-			manager := newManager("manager1", testJob, 10)
+	rc.LPush("prod:queue:manager1", message.ToJson()).Result()
+	rc.LPush("prod:queue:manager1", message2.ToJson()).Result()
 
-			rc.LPush("prod:queue:manager1", message.ToJson()).Result()
-			rc.LPush("prod:queue:manager1", message2.ToJson()).Result()
+	manager.start()
 
-			manager.start()
+	assert.Equal(t, message.Args(), <-processed)
+	assert.Equal(t, message2.Args(), <-processed)
 
-			c.Expect(<-processed, Equals, message.Args())
-			c.Expect(<-processed, Equals, message2.Args())
+	manager.quit()
 
-			manager.quit()
-
-			len, _ := rc.LLen("prod:queue:manager1").Result()
-			c.Expect(len, Equals, int64(0))
-		})
+	len, _ := rc.LLen("prod:queue:manager1").Result()
+	assert.Equal(t, int64(0), len)
+}
 
-		c.Specify("drain queue completely on exit", func() {
-			sentinel, _ := NewMsg("{\"foo\":\"bar2\",\"args\":\"sentinel\"}")
+func TestDrainQueueOnExit(t *testing.T) {
+	namespace := "prod"
+	setupTestConfigWithNamespace(namespace)
+	rc := Config.Client
 
-			drained := false
+	processed := make(chan *Args)
 
-			slowJob := (func(message *Msg) {
-				if message.ToJson() == sentinel.ToJson() {
-					drained = true
-				} else {
-					processed <- message.Args()
-				}
+	sentinel, _ := NewMsg("{\"foo\":\"bar2\",\"args\":\"sentinel\"}")
 
-				time.Sleep(1 * time.Second)
-			})
-			manager := newManager("manager1", slowJob, 10)
+	drained := false
 
-			for i := 0; i < 9; i++ {
-				rc.LPush("prod:queue:manager1", message.ToJson()).Result()
-			}
-			rc.LPush("prod:queue:manager1", sentinel.ToJson()).Result()
+	slowJob := (func(message *Msg) {
+		if message.ToJson() == sentinel.ToJson() {
+			drained = true
+		} else {
+			processed <- message.Args()
+		}
 
-			manager.start()
-			for i := 0; i < 9; i++ {
-				<-processed
-			}
-			manager.quit()
+		time.Sleep(1 * time.Second)
+	})
+	manager := newManager("manager1", slowJob, 10)
+
+	for i := 0; i < 9; i++ {
+		rc.LPush("prod:queue:manager1", message.ToJson()).Result()
+	}
+	rc.LPush("prod:queue:manager1", sentinel.ToJson()).Result()
+
+	manager.start()
+	for i := 0; i < 9; i++ {
+		<-processed
+	}
+	manager.quit()
+
+	len, _ := rc.LLen("prod:queue:manager1").Result()
+	assert.Equal(t, int64(0), len)
+	assert.True(t, drained)
+}
 
-			len, _ := rc.LLen("prod:queue:manager1").Result()
-			c.Expect(len, Equals, int64(0))
-			c.Expect(drained, Equals, true)
-		})
+func TestMultiMiddleware(t *testing.T) {
+	//per-manager middlwares are called separately, global middleware is called in each manager", func() {
```

Comment:
leftover comment?

Reply:
I would like to keep the comment but the `func` at the end needs to go.